### PR TITLE
17.17 - Search Views Index

### PR DESCRIPTION
**Pull request by diana-macrometa**
This PR rewrites the Search Views index so that it covers both Fulltext Search and Semantic Search.

### CODE REVIEW

**Review by elof — COMMENTED** [review]
Screenshot by Dropbox Capture

I got some errors for broken markdown links

File: docs/search-views/index.md [hunk 1]
```diff
@@ -3,3 +3,27 @@ sidebar_position: 1
...
+
+Search views in Macrometa are structured ways to query your global data network. They improve the speed and accuracy of data retrieval. Macrometa offers two types of search views: fulltext search and semantic search.
+
+## Fulltext Search
+
+Fulltext search is a method that searches for exact matches of terms or phrases in your query. It uses an index, which lists each word in the document and its location. Fulltext search is effective when the requirement is to find documents containing specific words or phrases.
+
+## Semantic Search
+
+Semantic search is a different approach that understands the intent behind a search query and the meanings of words. It uses natural language processing and machine learning techniques to interpret synonyms and the context of natural language queries. Semantic search returns results that are relevant to the query's intent rather than just literal matches.
```
Comment (amolth-macrometa) [inline docs/search-views/index.md]: While this is the correct definition and description of semantic search, we are only providing similarity search within this umbrella (no natural language queries or intent based search etc.), so it feels a little too grand. Maybe you can add that in 0.17.17 we are providing similarity search within this umbrella and that the objective is to be able to query for similar items within the items stored in GDN (or something like that).

File: docs/search-views/index.md [hunk 2]
```diff
@@ -3,3 +3,27 @@ sidebar_position: 1
...
+
+## Fulltext vs. Semantic Search Views
+
+Choosing between fulltext and semantic search views depends on your specific needs. Fulltext search is efficient for exact keyword searches. Semantic search is useful when understanding context, synonyms, and user intent matters.
```
Comment (amolth-macrometa) [inline docs/search-views/index.md]: we are not really doing the content/synonyms/intent in 17.17. (kind of same comment as above)

File: docs/search-views/index.md [hunk 3]
```diff
@@ -3,3 +3,27 @@ sidebar_position: 1
...
+
+| Criteria | Fulltext Search View | Semantic Search View |
+|--------------|--------------------------|--------------------------|
+| Type of search | Keyword-based | Contextual, based on intent |
+| Accuracy | High for exact keyword matches | High for contextual relevance |
+| Understanding synonyms | Does not match synonyms | Matches synonyms |
+| Natural language understanding | Limited | High |
+| Memory usage | Lower, stores keywords | Higher, stores vector representations of words |
```
Comment (amolth-macrometa) [inline docs/search-views/index.md]: similarity search stores not just similarity of words, but can also work with objects, images, videos etc.

File: docs/search-views/index.md [hunk 4]
```diff
@@ -3,3 +3,27 @@ sidebar_position: 1
...
+| Speed | Fast for exact matches | Can be slower due to complexity, but optimized with HNSW, IVF, and FLAT |
```
Comment (amolth-macrometa) [inline docs/search-views/index.md]: I think we don't yet have enough data for this. This anyway depends on the kind of parameters that we use. Not sure this line really provides value to the user.

**Review by amolth-macrometa — CHANGES_REQUESTED** [review]

**Comment by diana-macrometa** [comment]
> Screenshot by Dropbox Capture
> 
> I got some errors for broken markdown links

Yes, they are fixed in the main release branch. I'm in the middle of rearranging things for the new search.

File: docs/search-views/index.md (hunk 1 above)
Comment (diana-macrometa) [reply docs/search-views/index.md]: Updated. How's this?

**Comment by diana-macrometa** [comment]
I removed the comparison table. We can make a new one in future if it seems helpful.

**Comment by diana-macrometa** [comment]
I added some new content, please re-review.

**Review by maheshw2 — APPROVED** [review]

**Pull request merged**